ObjectiveGit/GTBranch.m

```diff
@@ -193,7 +193,12 @@ - (GTBranch *)trackingBranchWithError:(NSError **)error success:(BOOL *)success
 }
 
 - (BOOL)updateTrackingBranch:(GTBranch *)trackingBranch error:(NSError **)error {
-	int result = git_branch_set_upstream(self.reference.git_reference, trackingBranch.shortName.UTF8String);
+  int result = GIT_ENOTFOUND;
+  if (trackingBranch.branchType == GTBranchTypeRemote) {
+    result = git_branch_set_upstream(self.reference.git_reference, [trackingBranch.name stringByReplacingOccurrencesOfString:[GTBranch remoteNamePrefix] withString:@""].UTF8String);
```

Comment:
I'm not setting upstream for a remote branch. Oppositely, I'm setting local branch's upstream to a remote branch. I'm not quite sure why you thought this didn't make any sense. For a local branch, `shortName` can make underlying method to find out the correct refspec, but `shortName` doesn't work on remote refspec.

Not a native English speaker XD